broker/broker_schema_mutator.go

```diff
@@ -0,0 +1,69 @@
+package broker
+
+import (
+	"github.com/uber/aresdb/metastore/common"
+	"github.com/uber/aresdb/metastore"
+	"github.com/uber/aresdb/memstore"
+	"sync"
+)
+
+// BrokerSchemaMutator implements 2 interfaces
+//  - metastore.TableSchemaMutator needed by SchemaFetchJob
+//  - memstore.TableSchemaReader needed by compiler
+type BrokerSchemaMutator struct {
+	sync.RWMutex
+
+	tables map[string]*memstore.TableSchema
+}
+
+func NewBrokerSchemaMutator() *BrokerSchemaMutator {
+	return &BrokerSchemaMutator{}
+}
+
+// ====  metastore.TableSchemaMutator ====
+// TODO: implement. TableSchemaMutator should update b.tables
+func (b *BrokerSchemaMutator) ListTables() (tables []string, err error) {
+	return
+}
+
+func (b *BrokerSchemaMutator) GetTable(name string) (table *common.Table, err error) {
+	tableSchema, ok := b.tables[name]
+	if !ok {
+		err = metastore.ErrTableDoesNotExist
+	}
+	table = &tableSchema.Schema
+	return
+}
+
+func (b *BrokerSchemaMutator) CreateTable(table *common.Table) (err error) {
```

Comment:
since broker only get schema sync from the controller, we may not need to expose these functions

Reply:
this is needed as BrokerSchemaMutator struct implements TableSchemaMutator interface and mutates an in-memory map of tables.